Food items management

HafizHaris opened the pull request.

HafizHaris committed:
Merge pull request #3 from hmmdAhmd/theme-integration

Theme Integration

Commit:
Merge branch 'dev' of https://github.com/hmmdAhmd/Techlunch into dev

Commit:
Complete Food Item Management

Commit:
Merge remote-tracking branch 'origin/dev' into food-items-management

ts-nanwar commented on TechlunchApi/Controllers/FoodItemIngredientsController.cs:
```diff
@@ -0,0 +1,72 @@
+        // POST: api/FoodItemIngredients
+        // To protect from overposting attacks, enable the specific properties you want to bind to, for
+        [HttpPost]
+        public async Task<ActionResult<FoodItemIngredients>> PostFoodItemIngredients(FoodItemIngredients foodItemIngredients)
```
Lets change function name and built habit to follow coding standards.
https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions
https://www.c-sharpcorner.com/UploadFile/8a67c0/C-Sharp-coding-standards-and-naming-conventions/

ts-nanwar commented on TechlunchApi/Controllers/FoodItemIngredientsController.cs:
```diff
@@ -0,0 +1,72 @@
+                await _context.SaveChangesAsync();
+                return CreatedAtAction("GetFoodItemIngredients", new { id = foodItemIngredients.Id }, foodItemIngredients);
+            }
+            else {
```
Let's create separate APIs for the insert and update functions.

ts-nanwar commented on TechlunchApi/Controllers/FoodItemIngredientsController.cs:
```diff
@@ -0,0 +1,72 @@
+            {
+                return NotFound();
+            }
+            _context.FoodItemIngredients.Remove(foodItemIngredients);
```
I believe we're not working in sync. Hammad is using Delete to change the status and here we're deleting an actual record.

ts-nanwar commented on TechlunchApp/appsettings.json:
```diff
@@ -6,5 +6,6 @@
   },
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "apiUrl": "https://localhost:44357/api/"
```
Hammad had this change as well. So it should be part of your PR.

ts-nanwar commented on TechlunchApp/Controllers/FoodItemIngredientController.cs:
```diff
@@ -0,0 +1,83 @@
+            using (var httpClient = new HttpClient())
+            {
+                StringContent content = new StringContent(JsonConvert.SerializeObject(FoodItemIngredientObj), Encoding.UTF8, "application/json");
+                var response = await httpClient.PostAsync(_apiUrl + "FoodItemIngredients", content);
```
Check for best practice to concat string in C#

ts-nanwar replied:
Use it wherever it is applicable

ts-nanwar commented on TechlunchApp/Controllers/FoodItemController.cs:
```diff
@@ -0,0 +1,68 @@
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Delete(int ItemId)
```
Why Delete method is HTTPPost?

ts-nanwar commented on TechlunchApp/Controllers/FoodItemIngredientController.cs:
```diff
@@ -0,0 +1,83 @@
+        private readonly string _apiUrl;
+        public FoodItemIngredientController(IConfiguration configuration)
+        {
+            _apiUrl = configuration.GetValue<string>("apiUrl");
```
Please review Hammad PR comments as well.

Commit:
string script = "alert(\"Hello!\");";
ScriptManager.RegisterStartupScript(this, GetType(),
                      "ServerControlScript", script, true);

Commit:
Updated food item management

HafizHaris committed:
Merge branch 'dev' into food-items-management
